components/org.wso2.carbon.identity.api.server.common/src/main/java/org/wso2/carbon/identity/api/server/common/ContextLoader.java

```diff
@@ -73,9 +76,20 @@ public static String getUsernameFromContext() {
      */
     public static URI buildURIForBody(String endpoint) {
 
-        String tenantQualifiedRelativePath =
-                String.format(TENANT_CONTEXT_PATH_COMPONENT, getTenantDomainFromContext()) + SERVER_API_PATH_COMPONENT;
-        String url = IdentityUtil.getEndpointURIPath(tenantQualifiedRelativePath + endpoint, true, true);
+        String url = "";
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            String context = SERVER_API_PATH_COMPONENT + endpoint;
+            try {
+                url = ServiceURLBuilder.create().addPath(context).build().getRelativePublicURL();
+            } catch (URLBuilderException e) {
+                LOG.error("Server encountered an error while building the location URL in tenant qualified scheme.");
```

Comment:
I think we have to break the flow. If we cannot build the URL.
Check whether you can throw an APIError from here